types/src/error.rs

```diff
@@ -16,62 +18,110 @@ impl<T: fmt::Display> fmt::Display for Mismatch<T> {
 }
 
 /// Error type.
-#[derive(Debug, thiserror::Error)]
 pub enum Error {
 	/// Networking error or error on the low-level protocol layer.
-	#[error("Networking or low-level protocol error: {0}")]
-	TransportError(#[source] Box<dyn std::error::Error + Send + Sync>),
+	TransportError(Box<dyn fmt::Debug + Send + Sync>),
 	/// JSON-RPC request error.
-	#[error("JSON-RPC request error: {0:?}")]
-	Request(#[source] jsonrpc::Error),
+	Request(jsonrpc::Error),
 	/// Subscription error.
-	#[error("Subscription failed, subscribe_method: {0} unsubscribe_method: {1}")]
 	Subscription(String, String),
 	/// Frontend/backend channel error.
-	#[error("Frontend/backend channel error: {0}")]
-	Internal(#[source] futures::channel::mpsc::SendError),
+	#[cfg(feature = "std")]
+	Internal(futures::channel::mpsc::SendError),
 	/// Invalid response,
-	#[error("Invalid response: {0}")]
 	InvalidResponse(Mismatch<String>),
 	/// The background task has been terminated.
-	#[error("The background task been terminated because: {0}; restart required")]
 	RestartNeeded(String),
 	/// Failed to parse the data that the server sent back to us.
-	#[error("Parse error: {0}")]
-	ParseError(#[source] jsonrpc::ParseError),
+	ParseError(jsonrpc::ParseError),
 	/// Invalid subscription ID.
-	#[error("Invalid subscription ID")]
 	InvalidSubscriptionId,
 	/// Invalid request ID.
-	#[error("Invalid request ID")]
 	InvalidRequestId,
 	/// A request with the same request ID has already been registered.
-	#[error("A request with the same request ID has already been registered")]
 	DuplicateRequestId,
 	/// Method was already registered.
-	#[error("Method: {0} was already registered")]
 	MethodAlreadyRegistered(String),
 	/// Subscribe and unsubscribe method names are the same.
-	#[error("Cannot use the same method name for subscribe and unsubscribe, used: {0}")]
 	SubscriptionNameConflict(String),
 	/// Websocket request timeout
-	#[error("Websocket request timeout")]
 	WsRequestTimeout,
 	/// Configured max number of request slots exceeded.
-	#[error("Configured max number of request slots exceeded")]
 	MaxSlotsExceeded,
 	/// Custom error.
-	#[error("Custom error: {0}")]
 	Custom(String),
 }
 
+impl fmt::Debug for Error {
+	#[inline]
```

Comment:
Let's not try to outsmart the compiler; formatting errors is not on the hot path for anything so please remove the inlining hints.